app/models/user.php

```diff
@@ -0,0 +1,103 @@
+<?php
+class User extends AppModel
+{
+    const AUTH_USER_KEY = 'auth_user';
+
+    public $validation = array(
+        'username'      => array(
+            'length'    => array('validate_between', 1, 16),
+            'chars'     => array('validate_username'),
+            'unique'    => array('validate_unique_name')
+        ),
+        'email'         => array(
+            'length'    => array('validate_between', 1, 30),
+            'chars'     => array('validate_email')
+        ),
+        'password'      => array(
+            'length'    => array('validate_between', 6, 20))
+    );
+
+    public function create()
+    {
+        if(!$this->validate()) {
+            throw new ValidationException('Invalid user information');
+        }
+
+        $info = array(
+            'username'  => $this->username,
+            'email'     => $this->email,
+            'password'  => bhash($this->password)
+        );
+
+        $db = DB::conn();
+        $db->insert('user', $info);
```

Comment:
you could directly create the array inside $db->insert here po, no need to create an extra variable